composition/src/normalization/normalization-factory.ts

```diff
@@ -1352,32 +1363,61 @@ export class NormalizationFactory {
               isValid = false;
               continue;
             }
-            if (missingRequiredFieldNames.has(fieldName)) {
-              missingRequiredFieldNames.delete(fieldName);
+            if (allowedFieldNames.has(fieldName)) {
+              allowedFieldNames.delete(fieldName);
             } else {
-              duplicateRequiredFieldNames.add(fieldName);
+              duplicateFieldNames.add(fieldName);
               isValid = false;
               continue;
             }
-            if (field.value.kind !== Kind.STRING || field.value.value.length < 1) {
-              invalidRequiredFieldNames.add(fieldName);
-              isValid = false;
-              continue;
+            if (missingRequiredFieldNames.has(fieldName)) {
+              missingRequiredFieldNames.delete(fieldName);
             }
             switch (fieldName) {
               case CONSUMER_NAME:
+                if (field.value.kind != Kind.STRING || field.value.value.length < 1) {
+                  invalidRequiredFieldNames.add(fieldName);
+                  isValid = false;
+                  continue;
+                }
                 consumerName = field.value.value;
                 break;
               case STREAM_NAME:
+                if (field.value.kind != Kind.STRING || field.value.value.length < 1) {
+                  invalidRequiredFieldNames.add(fieldName);
+                  isValid = false;
+                  continue;
+                }
                 streamName = field.value.value;
                 break;
+              case CONSUMER_INACTIVE_THRESHOLD:
+                if (field.value.kind != Kind.INT) {
+                  errorMessages.push(
+                    invalidIntegerTypeValue('edfs__NatsStreamConfiguration(consumerInactiveThreshold: ...)'),
+                  );
+                  isValid = false;
+                  continue;
+                }
+
+                try {
+                  consumerInactiveThreshold = parseInt(field.value.value, 10);
+                } catch (e) {
+                  errorMessages.push(
+                    invalidIntegerValue(
+                      field.value.value,
+                      'edfs__NatsStreamConfiguration(consumerInactiveThreshold: ...)',
+                    ),
+                  );
+                  isValid = false;
+                }
+                break;
             }
           }
           if (!isValid || missingRequiredFieldNames.size > 0) {
             errorMessages.push(
               invalidNatsStreamInputFieldsErrorMessage(
                 [...missingRequiredFieldNames],
-                [...duplicateRequiredFieldNames],
+                [...duplicateFieldNames],
```

Comment:
just curious why the name change

Reply:
consumerInactiveThreshold is not required...

Reply:
It's required. It just defines a default value.

Reply:
If you set it to null, it would be an error. :D

Reply:
That is true, but I thought that in this context 'required' meant 'the argument is explicitly set.' That is what the code that checks for missingRequiredFieldNames is doing. I'm not in love with this name, so if you think the old name was better, I can keep it!

Reply:
it's fine